### PR TITLE
normalizing snarky yojson serializers to match kimchi common

### LINKED ISSUE #12591: Add yojson serializations to snarky
Types in snarky do not have yojson serializations defined for them (eg `Field`). In the mina daemon code, we have some custom implementations that provide yojson serializations for these types, but we should move them into the snarky layer instead.

### PR DESCRIPTION
**Pull request by svv232**
Closes #12591

### CODE REVIEW

**Comment by svv232** [comment]
!ci-build-me

File: src/lib/mina_wire_types/mina_base/mina_base_account_update.ml [hunk 1]
```diff
@@ -2,6 +2,7 @@ module Authorization_kind = struct
   module V1 = struct
     (* field for Proof is a verification key hash *)
     type t = Signature | Proof of Snark_params.Tick.Field.t | None_given
+    [@@deriving yojson]
```
Comment (mrmr1993) [inline src/lib/mina_wire_types/mina_base/mina_base_account_update.ml:5]: Please don't add this here; the `wire_types` library should only contain types

**Review by mrmr1993 — CHANGES_REQUESTED** [review]

File: src/lib/mina_wire_types/mina_base/mina_base_account_update.ml (hunk 1 above)
Comment (svv232) [reply src/lib/mina_wire_types/mina_base/mina_base_account_update.ml]: i've reimplemented this solution to not do this

**Comment by svv232** [comment]
!ci-build-me

**Comment by svv232** [comment]
!ci-build-me

**Comment by svv232** [comment]
!ci-build-me

**Review by nholland94 — APPROVED** [review]

**Comment by svv232** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Pull request merged**